### PR TITLE
Stabilize fuzz program & json

### PR DESCRIPTION
**Pull request by Juan-M-V**
# Implement arbitrary by hand for some structs

## Description

This PR aims to increase the edge coverage of `fuzz_json` and `fuzz_program` by implementing arbitrary by hand in some of the internal structs like `Location`. Also words were added to the `dict.json` to include some common tokens present in `.json` programs, this last change will be temporary as a better strategy is developed

## Checklist
- [ ] Linked to Github Issue
- [ ] Unit tests added
- [ ] Integration tests added.
- [ ] This change requires new documentation.
  - [ ] Documentation has been added/updated.
  - [ ] CHANGELOG has been updated.

### CODE REVIEW

File: fuzzer/src/fuzz_json.rs
Comment (Juan-M-V) [inline fuzzer/src/fuzz_json.rs:11]: Using `cairo_run` instead of `cairo_run_fuzzed_program` due to the unlikeliness of the fuzzer creating an infinite loop and wanting to get as close to normal VM execution as possible

**Review by pefontana — APPROVED** [review]

File: vm/src/serde/deserialize_program.rs
Comment (Oppen) [inline vm/src/serde/deserialize_program.rs:156]: Maybe remove this comment.

File: vm/src/serde/deserialize_program.rs
Comment (Oppen) [inline vm/src/serde/deserialize_program.rs:197]: This essentially converts the location vector into a linked list, right? Is this the intended behavior?

File: vm/src/types/program.rs
Comment (Oppen) [inline vm/src/types/program.rs:72]: We should probably find a way to bias towards:
1. `Felt252`, as `Relocatable` should not be allowed (and indeed we should change it in the struct);
2. Values in the range `[0..2^63)`, so it's more likely to create instructions than random data that will be rejected.
Maybe this can be done with an alternate function taking weights for those?

**Review by Oppen — COMMENTED** [review]
I left one question and a suggestion. Overall looks good, waiting on the answer for approval.

File: vm/src/serde/deserialize_program.rs
Comment (Juan-M-V) [reply vm/src/serde/deserialize_program.rs:197]: Yes, it is. Though I want to improve the `Location` generation, I mainly wanted arbitrary to stop crashing, as it currently does with recursive structures.

File: vm/src/types/program.rs
Comment (Juan-M-V) [reply vm/src/types/program.rs]: how about: https://github.com/lambdaclass/cairo-vm/pull/1344/commits/732f9f908e427e92a737ce0b721fdf81c73e1916
```rust
let mut data = Vec::new(); //arbitrary(u)?;
for _ in 0..usize::arbitrary(u)? {
    let instruction = u64::arbitrary(u)?;
    data.push(MaybeRelocatable::from(Felt252::from(instruction)));
    // Check if the Imm falg is on and add an immediate value if it is
    if instruction & 0x0004000000000000 != 0 {
        data.push(MaybeRelocatable::from(Felt252::arbitrary(u)?));
    }
}
```

Comment (Juan-M-V) [reply vm/src/types/program.rs]: I'm not too sure how to avoid a magic number `if instruction & 0x0004000000000000 != 0 {` while also reducing overhead for generating teh data vector

File: vm/src/types/program.rs
Comment (Oppen) [inline vm/src/types/program.rs:76]:
```suggestion
            // Check if the Imm flag is on and add an immediate value if it is
```

File: vm/src/types/program.rs
Comment (Oppen) [reply vm/src/types/program.rs]: I think that's a good solution.

Comment (Oppen) [reply vm/src/types/program.rs]: With a caveat: an interesting case to test is where the last instruction expects an immediate but there's none. So maybe something like:
```rust
let (mut i, len) = (0, u64::arbitrary(u)?);
while i < len {
    i += 1;
    // generate instruction
    if i < len && instruction & 0x0004000000000000 != 0 {
         // generate Felt
         i += 1;
    }
}
```

I think that should cover all interesting cases?

Comment (Juan-M-V) [reply vm/src/types/program.rs]: Similar code: https://github.com/lambdaclass/cairo-vm/pull/1344/commits/240726a40667f66730a08deb1a2b8582f1459c6e
I trust `for i in ...` more than `while`, less likely to forget the `i += 1`

**Review by Oppen — APPROVED** [review]

**Pull request merged**
